src/plugins/crypto.c

```diff
@@ -1073,7 +1073,7 @@ gboolean bd_crypto_tc_open_full (const gchar *device, const gchar *name, const g
 
     if (keyfiles) {
         for (i=0; *(keyfiles + i); i++) {
-            keyfiles_count = i;
+            keyfiles_count = i + 1;
```

Comment:
Why is ``keyfiles_count`` not assigned just once to ``i`` after the cycle?

Reply:
Yeah, to me that would look more like counting. This looks to me like there is something missing in the loop that should use the updated value of ``keyfiles_count``.